dlpy/embedding_model.py

```diff
@@ -0,0 +1,424 @@
+                                 seed=seed, record_seed=record_seed,
+                                 force_equal_padding=True,
+                                 save_best_weights=save_best_weights, n_threads=n_threads,
+                                 target_order=None, train_from_scratch=train_from_scratch_real)
+                res.append(res_t)
+                # drop this data
+                data.droptable()
+            print('Note: Training with data generation took {} (s)'.format(time.time() - time_start))
+        return res
+
+    def deploy_embedding_model(self, path, output_format='astore', model_type='branch'):
+        """
+        Deploy the deep learning model to a data file
+
+        Parameters
+        ----------
+        path : string
+            Specifies the location to store the model files.
+            If the output_format is set to castable, then the location has to be on the server-side.
+            Otherwise, the location has to be on the client-side.
+        output_format : string, optional
+            Specifies the format of the deployed model.
+            When astore is specified, the learned embedding features will be output as well.
+            Valid Values: astore, castable, or onnx
+            Default: astore
+        model_type : string, optional
+            Specifies how to deploy the embedding model. "branch" means only one branch model is deployed to extract
+            features while "full" means the full model is deployed to extract features for all branches and
+            compute the distance metric values for all input data pairs.
+            Valid values: branch and full
+            Default: branch
+
+        Notes
+        -----
+        Currently, this function supports sashdat, astore, and onnx formats.
+
+        More information about ONNX can be found at: https://onnx.ai/
+
+        DLPy supports ONNX version >= 1.3.0, and Opset version 8.
+
+        For ONNX format, currently supported layers are convo, pool,
+        fc, batchnorm, residual, concat, reshape, and detection.
+
+        If dropout is specified in the model, train the model using
+        inverted dropout, which can be specified in :class:`Optimizer`.
+        This will ensure the results are correct when running the
+        model during test phase.
+
+        Returns
+        --------
+        :class:`Model` for a branch model when model_type is 'branch'
+
+        """
+
+        if model_type.lower() not in ['branch', 'full']:
+            raise DLPyError('Only branch and full are valid.')
+
+        if model_type.lower() == 'full':
+            temp_embed_layers = []
+            for i_branch in range(self.number_of_branches):
+                temp_embed_layers.append(self.embedding_layer_name_prefix + str(i_branch))
+            self.deploy(path=path, output_format=output_format, layers=temp_embed_layers)
+        else:
+            # create a fake task layer
```

Comment:
It would be better to inform our users that we add a classification layer after the branch.

Reply:
I would like to keep this internal. I tried to remove the task layer but our attr functions to change model attrs assume a classification model. When we get time to clean up attr functions, we could remove this fake layer issue.